src/main/java/com/example/myKitchenManager/MyKitchenManagerController.java

```diff
@@ -19,4 +23,12 @@ public String welcome () {
     public String login () {
         return "This is a page for user to log in";
     }
+
+    @RequestMapping("/ingredient")
+    public List<String> ingredient() {
```

Comment:
Would be better if we use the Controller-Modle pattern, means we have the endpoint for ingredient and we have the ingredient object that mapping to the ingredient table in the DB. This pattern will made CRUD of ingredient much easier. https://www.javaguides.net/2019/01/standard-project-structure-for-spring-boot-projects.html introduce some good code structure we can follow.

Reply:
Since in this controller layer what we usually do is handling request. For example of there are parameters in the ingradient request like /getIngradient?id=..., or we want to get the list of a user /getIngradients?uuid=xxx, we need to first  check the authentication of the request (If the user is login or not), and we need to gee the parameters in the request. 

Then we will have an object like "ingradientManager" this will handle the function like ingradientManager.findByID() or ingradientManager.findByUser(). These will return us an object of ingradient or list of ingradients, the ingradient object is just mapping to the ingradient table. Please check "JAVA Hibernate" for this part.

After we get the object of ingradient or list of ingradients, we can prepare the Response in this controller.